openstax/api.py

```diff
@@ -1,229 +1,20 @@
-
-from django.utils import timezone
-
-from .functions import remove_locked_links_listing, remove_locked_links_detail
-
-from flags.state import flag_state
-
-import json
-
-
-# Create the router. "wagtailapi" is the URL namespace
+# # Create the router. "wagtailapi" is the URL namespace
 api_router = WagtailAPIRouter('wagtailapi')
 admin_api = WagtailAPIRouter('wagtailadmin_api_v1')
 
-#TODO: This is fixed in an upcoming version of Wagtail, and should be removed.
-# If the API is causing errors after an upgrade - this is likely the first two classes to remove!
-class ImagesAPIEndpoint(ImagesAPIEndpoint):
-
-    body_fields = ImagesAPIEndpoint.body_fields + [
-        'file',
-    ]
-
-    listing_default_fields = ImagesAPIEndpoint.listing_default_fields + [
-        'file',
-    ]
-
-class DocumentsAPIEndpoint(DocumentsAPIEndpoint):
-    body_fields = DocumentsAPIEndpoint.body_fields + [
-        'file',
-    ]
-
-    listing_default_fields = DocumentsAPIEndpoint.listing_default_fields + [
-        'file',
-    ]
-
-        
-class PagesAPIEndpoint(PagesAPIEndpoint):
-    def listing_view(self, request):
-
-        response = super().listing_view(request)
-        
-        # Implementing User Authentication
-        auth_user = json.loads(get_user_data(request).content.decode())
-
-        # Fetching User Information and Overwriting auth_user
-        if "user" in auth_user:
-            auth_user = get_user_info(auth_user["user"]["id"])
-        
-        # Overwriting the Response if ox credential does not
-        # authorize faculty access.
-        if "faculty_status" not in auth_user or auth_user["faculty_status"] != "confirmed_faculty":
-            if flag_state('hide_faculty_resources', bool=True):
-                remove_locked_links_listing(response)
-
-        return response
-
-    def detail_view(self, request, pk):
-
-        any_hidden = False
-
-        response = super().detail_view(request, pk)
-        page = Page.objects.get(pk=pk)
-
-        # Implementing User Authentication
-        auth_user = json.loads(get_user_data(request).content.decode())
-
-        # Fetching User Information and Overwriting auth_user
-        if "user" in auth_user:
-            auth_user = get_user_info(auth_user["user"]["id"])
-
-        # Overwriting the Response if ox credential does not
-        # authorize faculty access.
-        if "faculty_status" not in auth_user or auth_user["faculty_status"] != "confirmed_faculty":
-            if flag_state('hide_faculty_resources', bool=True):
-                any_hidden = remove_locked_links_detail(response)
-
-        # Implementing Caching
-        response['Cache-Control'] = 'no-cache'
-
-        # If we ended up revealing a link, then force the content to be loaded.
-        if not any_hidden or request.GET.get('force-reload'):
-            response['Last-Modified'] = timezone.now()
-
-        
-        return response
-
-
-## TODO: This can be removed when Wagtail fixes admin API
```

Comment:
@edwoodward - I think everything above this line needs to be kept here. Overriding the image, document, and page APIs are still required.

Reply:
@mwharrison Just so I understand, how did you determine this? I did not see any errors when it was removed.

Reply:
Fixed as we discussed